service/src/parentchain_config.rs

```diff
@@ -0,0 +1,140 @@
+};
+
+/// Standard runtime config for Substrate and Polkadot nodes.
+pub type IntegriteeRuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, integritee::ParentchainExtrinsicParams>;
+
+/// runtime config for chains like Asset Hub or Encointer
+pub type TargetARuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, target_a::ParentchainExtrinsicParams>;
+
+/// Standard runtime config for Substrate and Polkadot nodes.
+pub type TargetBRuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, target_b::ParentchainExtrinsicParams>;
+
+pub enum ParentchainApiLocal {
+	Integritee(IntegriteeParentchainApi),
+	TargetA(TargetAParentchainApi),
+	TargetB(TargetBParentchainApi),
+}
+
+impl ParentchainApiLocal {
+	pub fn get_opaque_storage_by_key(
+		&self,
+		storage_key: StorageKey,
+		at_block: Option<Hash>,
+	) -> ApiResult<Option<Vec<u8>>> {
+		match self {
+			Self::Integritee(api) => api.get_opaque_storage_by_key(storage_key, at_block),
+			Self::TargetA(api) => api.get_opaque_storage_by_key(storage_key, at_block),
+			Self::TargetB(api) => api.get_opaque_storage_by_key(storage_key, at_block),
+		}
+	}
+	pub fn get_storage_proof_by_keys(
+		&self,
+		storage_keys: Vec<StorageKey>,
+		at_block: Option<Hash>,
+	) -> ApiResult<Option<ReadProof<Hash>>> {
+		match self {
+			Self::Integritee(api) => api.get_storage_proof_by_keys(storage_keys, at_block),
+			Self::TargetA(api) => api.get_storage_proof_by_keys(storage_keys, at_block),
+			Self::TargetB(api) => api.get_storage_proof_by_keys(storage_keys, at_block),
+		}
+	}
+	pub fn submit_and_watch_opaque_extrinsic_until(
+		&self,
+		encoded_extrinsic: &Bytes,
+		watch_until: XtStatus,
+	) -> ApiResult<ExtrinsicReport<Hash>> {
+		match self {
+			Self::Integritee(api) =>
+				api.submit_and_watch_opaque_extrinsic_until(encoded_extrinsic, watch_until),
+			Self::TargetA(api) =>
+				api.submit_and_watch_opaque_extrinsic_until(encoded_extrinsic, watch_until),
+			Self::TargetB(api) =>
+				api.submit_and_watch_opaque_extrinsic_until(encoded_extrinsic, watch_until),
+		}
+	}
+	pub fn submit_opaque_extrinsic(&self, encoded_extrinsic: &Bytes) -> ApiResult<Hash> {
+		match self {
+			Self::Integritee(api) => api.submit_opaque_extrinsic(encoded_extrinsic),
+			Self::TargetA(api) => api.submit_opaque_extrinsic(encoded_extrinsic),
+			Self::TargetB(api) => api.submit_opaque_extrinsic(encoded_extrinsic),
+		}
+	}
+}
+
+pub type IntegriteeParentchainApi = Api<IntegriteeRuntimeConfig, TungsteniteRpcClient>;
+pub type TargetAParentchainApi = Api<TargetARuntimeConfig, TungsteniteRpcClient>;
+pub type TargetBParentchainApi = Api<TargetBRuntimeConfig, TungsteniteRpcClient>;
+
+pub struct IntegriteeParentchainApiWrapper(Api<IntegriteeRuntimeConfig, TungsteniteRpcClient>);
+
+// fixme: this is an ugly hack because api-client's 'new()' isn't part of any trait we could implement therefore we can't use the fn new() on generic api types
+impl ParentchainApiWrapper for IntegriteeParentchainApiWrapper {
```

Comment:
really ugly. there has to be a better way. but we may need to patch api-client

Reply:
Wrapper not necessary, we can just define our own trait and implement it on the api client IMO, which is a bit better.